### PR TITLE
2.0.0 dev Device history

### PR DESCRIPTION
**Pull request by jontore**
Possible to get device history.

### CODE REVIEW

File: entities/Device.js [hunk 1]
```diff
@@ -13,6 +15,7 @@ default class Device {
         this.owner = config.owner;
         this.openToPublic = config.public;
         this.ajax = new Ajax(config.ajax);
+        this.history = new DeviceHistory(config);
```
Comment (SmbatYeranyan) [inline entities/Device.js]: config is being passed around quite alot, im wondering if its best to make it a shared variable between classes or perhaps a configService class.

File: test/entities/history/deviceHistory.test.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                });
+
+            });
+            describe('on failure', function() {
+                it('should reject the promise with error message', function(done) {
+                    deviceHistoryInstance.getHistoricalData({
+                        periode: '1m'
+                    }).then(() => {}, (m) => {
+                        var obj = JSON.parse(m.response);
+                        expect(obj.message).to.equal('oh noes');
```
Comment (SmbatYeranyan) [inline test/entities/history/deviceHistory.test.js]: oh noes :)

File: tools/ajax.js [hunk 3]
```diff
@@ -7,10 +7,16 @@ default class Ajax {
         this.customXHR;
     }
 
+    get(url, raw, queryObj) {
```
Comment (SmbatYeranyan) [inline tools/ajax.js]: url, raw queryObj on my model branch I needed contentType. So queryObj will conflict. Perhaps we should compress these parameters? maybe just get(opts = {})? We would need to refactor few areas

**Comment by SmbatYeranyan** [comment]
Overall looks good 👍  for merge, but lets discuss the ajax part offline.

File: tools/ajax.js (hunk 3 above)
Comment (jontore) [inline tools/ajax.js]: Yes, that would be a better idea.

**Pull request merged**